### PR TITLE
Extract app and keys

### PR DESCRIPTION
**Pull request by GasimGasimzada**
- Extract app component from entrypoint
- Move canvas and roughjs as reference within the app
- Remove ReactDOM double rendering
- Use refs to refer to canvas and roughjs context instead of global variables
- Extract Keys and Key related utilities into their own modules

### CODE REVIEW

**Comment by vjeux** [comment]
Could you revert the rename so that github diff is actually useful?

**Comment by GasimGasimzada** [comment]
Done!

File: src/index.tsx [hunk 1]
```diff
@@ -1108,7 +1096,3 @@ class App extends React.Component<{}, AppState> {
 
 const rootElement = document.getElementById("root");
 ReactDOM.render(<App />, rootElement);
-const canvas = document.getElementById("canvas") as HTMLCanvasElement;
```
Comment (vjeux) [inline src/index.tsx]: Nice!

**Comment by vjeux** [comment]
Feel free to rename after you land this

**Pull request merged**